.ci/google-java-format.sh

```diff
@@ -25,7 +25,7 @@ INPUT_PATHS=($(find src/it/resources/com/google/checkstyle/test/ -name "Input*.j
     | grep -v "rule331nowildcard" \
     | grep -v "rule332nolinewrap" \
     | grep -v "rule333orderingandspacing" \
-    | grep -v "rule3421overloadsplit" \
+    | grep -v "rule3421overloadsplit/InputOverloadsNeverSplit.java" \
```

Comment:
diff for grepped file:

```diff
C:\OpenSource\checkstyle git:[15449-rule3421overloadsplit]
git diff src/it/resources/com/google/checkstyle/test/chapter3filestructure/rule3421overloadsplit/InputOverloadsNeverSplit.java
diff --git a/src/it/resources/com/google/checkstyle/test/chapter3filestructure/rule3421overloadsplit/InputOverloadsNeverSplit.java b/src/it/resources/com/google/checkstyle/test/chapter3filestructure/rule3421overloadsplit/InputOverloadsNeverSplit.java
index 4a2aff594..68360a1ba 100644
--- a/src/it/resources/com/google/checkstyle/test/chapter3filestructure/rule3421overloadsplit/InputOverloadsNeverSplit.java
+++ b/src/it/resources/com/google/checkstyle/test/chapter3filestructure/rule3421overloadsplit/InputOverloadsNeverSplit.java
@@ -1,8 +1,6 @@
package com.google.checkstyle.test.chapter3filestructure.rule3421overloadsplit;
 
-/**
- * Testing.
- */
+/** Testing. */
 public class InputOverloadsNeverSplit {
 
   int abc;
@@ -18,15 +16,16 @@ public class InputOverloadsNeverSplit {
   void foo2() {}
 
   InputOverloadsNeverSplit(int a) {}
+
   // violation above 'Constructors should be grouped together.*'
 
   int abcd;
 
   InputOverloadsNeverSplit(double x) {}
+
   // violation above 'Constructors should be grouped together.*'
 
   private enum InnerEnum1 {
-
     one;
 
     int xyz;
@@ -55,9 +54,11 @@ public class InputOverloadsNeverSplit {
   }
 
   InputOverloadsNeverSplit(float x) {}
+
   // violation above 'Constructors should be grouped together.*'
 
   InputOverloadsNeverSplit(long l) {}
+
   // violation above 'Constructors should be grouped together.*'
 
   private class Inner {
@@ -87,18 +88,23 @@ public class InputOverloadsNeverSplit {
   }

   InputOverloadsNeverSplit(long l, double d) {}
+
   // violation above 'Constructors should be grouped together.*'

-  InputOverloadsNeverSplit annoynmous = new InputOverloadsNeverSplit() {
-    int xyz;
+  InputOverloadsNeverSplit annoynmous =
+      new InputOverloadsNeverSplit() {
+        int xyz;

-    void test() {}
+        void test() {}

-    void test2() {}
-  };
+        void test2() {}
+      };

   private enum InnerEnum2 {
-    ONE, TWO, THREE;
+    ONE,
+    TWO,
+    THREE;
+
     void test() {}

     void test2() {}
@@ -107,7 +113,7 @@ public class InputOverloadsNeverSplit {
   }

   private enum InnerEnum3 {
-      InnerEnum3() {}
+    InnerEnum3() {}
   }

   private enum InnerEnum4 {}
@@ -127,63 +133,55 @@ public class InputOverloadsNeverSplit {
   private class Inner5 {}

   public void overloadMethod(int i) {
-    //some foo code
+    // some foo code
   }

   public void overloadMethod(String s) {
-    //some foo code
+    // some foo code
   }

   public void overloadMethod(boolean b) {
-    //some foo code
+    // some foo code
   }

-  public void fooMethod() {
-
-  }
+  public void fooMethod() {}

   // violation below 'All overloaded methods should be placed next to each other. .* '137'.'
   public void overloadMethod(String s, Boolean b, int i) {
-    //some foo code
+    // some foo code
   }

-  InputOverloadsNeverSplit anonymous = new InputOverloadsNeverSplit() {
-    public void overloadMethod(int i) {
-      //some foo code
-    }
+  InputOverloadsNeverSplit anonymous =
+      new InputOverloadsNeverSplit() {
+        public void overloadMethod(int i) {
+          // some foo code
+        }

-    public void overloadMethod(String s) {
-      //some foo code
-    }
+        public void overloadMethod(String s) {
+          // some foo code
+        }

-    public void overloadMethod(boolean b) {
-      //some foo code
-    }
+        public void overloadMethod(boolean b) {
+          // some foo code
+        }

-    public void fooMethod() {
+        public void fooMethod() {}

-    }
-
-    // violation below 'All overloaded methods should be placed next to each other. .* '159'.'
-    public void overloadMethod(String s, Boolean b, int i) {
-      //some foo code
-    }
-  };
+        // violation below 'All overloaded methods should be placed next to each other. .* '159'.'
+        public void overloadMethod(String s, Boolean b, int i) {
+          // some foo code
+        }
+      };

-  public void testing() {
-  }
+  public void testing() {}

-  private void testing(int a) {
-  }
+  private void testing(int a) {}

-  public void testing(int a, int b) {
-  }
+  public void testing(int a, int b) {}

-  public static void testing(String a) {
-  }
+  public static void testing(String a) {}

-  public void testing(String a, String b) {
-  }
+  public void testing(String a, String b) {}

   interface Fooable {
     public abstract void foo(int i);
@@ -217,24 +215,22 @@ public class InputOverloadsNeverSplit {
     }

     public void overloadMethod(int i) {
-      //some foo code
+      // some foo code
     }

     public void overloadMethod(String s) {
-      //some foo code
+      // some foo code
     }

     public void overloadMethod(boolean b) {
-      //some foo code
+      // some foo code
     }

-    public void fooMethod() {
-
-    }
+    public void fooMethod() {}

     // violation below 'All overloaded methods should be placed next to each other. .* '227'.'
     public void overloadMethod(String s, Boolean b, int i) {
-      //some foo code
+      // some foo code
     }
   }
 }
```

There were blank lines added between violation above comments and violation line, that was causing test to fail

Reply:
What was error of Junit? It looks like this diff should not affect target validation.

Reply:
formatted input:

```java
package com.google.checkstyle.test.chapter3filestructure.rule3421overloadsplit;

/** Testing. */
public class InputOverloadsNeverSplit {

  int abc;

  int bcd;

  void foo() {}

  InputOverloadsNeverSplit() {}

  InputOverloadsNeverSplit(String a) {}

  void foo2() {}

  InputOverloadsNeverSplit(int a) {}

  // violation above 'Constructors should be grouped together.*'

  int abcd;

  InputOverloadsNeverSplit(double x) {}

  // violation above 'Constructors should be grouped together.*'

  private enum InnerEnum1 {
    one;

    int xyz;

    InnerEnum1() {}

    InnerEnum1(String f) {}

    String str;

    String str2;

    InnerEnum1(int x) {} // violation 'Constructors should be grouped together.*'

    private abstract class Inner {
      Inner() {}

      int xyz;

      String neko;

      Inner(String g) {} // violation 'Constructors should be grouped together.*'
    }

    InnerEnum1(double d) {} // violation 'Constructors should be grouped together.*'
  }

  InputOverloadsNeverSplit(float x) {}

  // violation above 'Constructors should be grouped together.*'

  InputOverloadsNeverSplit(long l) {}

  // violation above 'Constructors should be grouped together.*'

  private class Inner {
    Inner() {}

    Inner(String str) {}

    // Comments are allowed between constructors.
    Inner(int x) {}
  }

  private class Inner2 {
    Inner2() {}

    Inner2(String str) {}

    int xyz;

    Inner2(int x) {} // violation 'Constructors should be grouped together.*'

    String xx;

    Inner2(double d) {} // violation 'Constructors should be grouped together.*'

    // violation below 'Constructors should be grouped together.*'
    Inner2(float f) {}
  }

  InputOverloadsNeverSplit(long l, double d) {}

  // violation above 'Constructors should be grouped together.*'

  InputOverloadsNeverSplit annoynmous =
      new InputOverloadsNeverSplit() {
        int xyz;

        void test() {}

        void test2() {}
      };

  private enum InnerEnum2 {
    ONE,
    TWO,
    THREE;

    void test() {}

    void test2() {}

    void test3() {}
  }

  private enum InnerEnum3 {
    InnerEnum3() {}
  }

  private enum InnerEnum4 {}

  private class Inner3 {
    void test() {}

    void test2() {}

    void test3() {}
  }

  private class Inner4 {
    Inner4() {}
  }

  private class Inner5 {}

  public void overloadMethod(int i) {
    // some foo code
  }

  public void overloadMethod(String s) {
    // some foo code
  }

  public void overloadMethod(boolean b) {
    // some foo code
  }

  public void fooMethod() {}

  // violation below 'All overloaded methods should be placed next to each other. .* '137'.'
  public void overloadMethod(String s, Boolean b, int i) {
    // some foo code
  }

  InputOverloadsNeverSplit anonymous =
      new InputOverloadsNeverSplit() {
        public void overloadMethod(int i) {
          // some foo code
        }

        public void overloadMethod(String s) {
          // some foo code
        }

        public void overloadMethod(boolean b) {
          // some foo code
        }

        public void fooMethod() {}

        // violation below 'All overloaded methods should be placed next to each other. .* '159'.'
        public void overloadMethod(String s, Boolean b, int i) {
          // some foo code
        }
      };

  public void testing() {}

  private void testing(int a) {}

  public void testing(int a, int b) {}

  public static void testing(String a) {}

  public void testing(String a, String b) {}

  interface Fooable {
    public abstract void foo(int i);

    public abstract void foo(String s);

    public abstract void noFoo();

    public abstract void foo(String s, Boolean b, int i);
    // violation above 'All overloaded methods should be placed next to each other. .* '191'.'

  }

  enum FooType {
    Strategy(""),
    Shooter(""),
    RPG("");

    private String description;

    private FooType(String description) {
      this.description = description;
    }

    public String getDescription() {
      return description;
    }

    public void setDescription(String description) {
      this.description = description;
    }

    public void overloadMethod(int i) {
      // some foo code
    }

    public void overloadMethod(String s) {
      // some foo code
    }

    public void overloadMethod(boolean b) {
      // some foo code
    }

    public void fooMethod() {}

    // violation below 'All overloaded methods should be placed next to each other. .* '227'.'
    public void overloadMethod(String s, Boolean b, int i) {
      // some foo code
    }
  }
}
```

Test result:

```

Violation lines for C:\OpenSource\checkstyle\src\it\resources\com\google\checkstyle\test\chapter3filestructure\rule3421overloadsplit\InputOverloadsNeverSplit.java differ.
missing (5)   : 19, 25, 57, 61, 91
unexpected (5): 18, 24, 56, 60, 90
---
expected      : [19, 25, 41, 50, 53, 57, 61, 80, 84, 87, 91, 150, 171, 193, 232]
but was       : [18, 24, 41, 50, 53, 56, 60, 80, 84, 87, 90, 150, 171, 193, 232]
```

failure cause:

```java
  InputOverloadsNeverSplit(double x) {}

  // violation above 'Constructors should be grouped together.*'
```

formatter formats the javadoc comments and separates it from the violation line.

we could not make it single liner like:


```java
  InputOverloadsNeverSplit(double x) {}  // violation 'Constructors should be grouped together.*'
```

because this would cross 100 width column limit

Reply:
nothing should stop you to do:
```
   InputOverloadsNeverSplit(int a) {}
-  // violation above 'Constructors should be grouped together.*'
+
+  // violation 2 lines above 'Constructors should be grouped together.*'
 
```

done.

Reply:
we do not care about such extra lines.